pkg/download/download.go

```diff
@@ -82,29 +83,32 @@ func Download(endpoint string, opts ...Option) (b []byte, err error) {
 			return retry.ExpectedError(err)
 		}
 
-		if dlOpts.Format == b64 {
-			var b64 []byte
-
-			b64, err = base64.StdEncoding.DecodeString(string(b))
-			if err != nil {
-				return err
-			}
-
-			b = b64
-		}
-
 		return nil
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to download config from %q: %w", u.String(), err)
 	}
 
+	if dlOpts.Format == b64 {
+		var b64 []byte
+
+		b64, err = base64.StdEncoding.DecodeString(string(b))
+		if err != nil {
+			return nil, err
+		}
+
+		b = b64
+	}
+
 	return b, nil
 }
 
-// download handles the actual http request
 func download(req *http.Request) (data []byte, err error) {
-	client := &http.Client{}
+	transport := (http.DefaultTransport.(*http.Transport))
```

Comment:
this is not a copy, it's a reference to default transport, so it has a side effect of installing `tftp` to the default transport